graphium/trainer/predictor.py

```diff
@@ -290,10 +297,16 @@ def compute_loss(
 
         if weights is not None:
             raise NotImplementedError("Weights are no longer supported in the loss")
-        all_task_losses = {
-            task: wrapped(preds=preds[task], target=targets[task])
-            for task, wrapped in wrapped_loss_fun_dict.items()
-        }
+
+        # Get the loss for each task
+        all_task_losses = {}
+        for task, wrapped in wrapped_loss_fun_dict.items():
+            this_loss = wrapped(preds=preds[task], target=targets[task])
+            if loss_empty_factor is not None:
+                this_loss = this_loss * loss_empty_factor[task]
+            all_task_losses[task] = this_loss
+
```

Comment:
We have access to the `all_task_losses` here with size of [gradient_acc*replica*device_iter], why not filter out zeros here directly?

Reply:
I think your right. We can do the filtering here. 

Also,  when the loss is 0 for a task, it gives 0 gradient only for the given task head, other task heads still have gradient.

Even if that task affects les the global gradient of the model, it is the only one affecting its own task head, which should get updated accordingly. 

So perhaps we don't need to adjust the loss the way that I've written it, and we can just keep normal behaviour for the gradient, but for the logging, we filter out 0- losses